src/Wyam/Program.cs

```diff
@@ -394,6 +396,10 @@ private Engine GetEngine()
                 {
                     engine.CleanOutputPathOnExecute = false;
                 }
+                if (_engineGlobalMetadata != null)
+                {
+                    engine.GlobalMetadata = _engineGlobalMetadata;
```

Comment:
That's a very good catch. Conceptually, we want the global metadata to start at the same point for each generation because any changes that get made will be "replayed" for following generations. If it persisted, our generations might end up polluting each other. I'd like to leave the global metadata writable from the config file too (it's already read-only from the execution context). That leaves some sort of cloning operation for each new engine.

`SimpleMetadata` is literally just a `Dictionary<string, object>`, so what if we added `ISimpleMetadata.Clone()` (which shouldn't be hard to write) and then made the line above:
```
engine.GlobalMetadata = _engineGlobalMetadata.Clone();
```